app/src/test/java/mozilla/lockbox/mocks/MockLoginsStorage.kt

```diff
@@ -43,6 +46,30 @@ open class MockLoginsStorage : LoginsStorage {
             SyncResult.fromValue(true)
         }
     }
+    override fun add(login: ServerPassword): SyncResult<String> {
+        return list().then {
+            if (login.id == "") {
+                SyncResult.fromException<String>(InvalidRecordException("record id missing"))
+            } else if (null != it.find { it.id == login.id }) {
+                SyncResult.fromException<String>(IdCollisionException("record already exists"))
+            }
```

Comment:
This logic needs a little bit of re-work. 

Can we use `get(login.id).then {}`

Can we put the empty string check as a guard condition at the top of the method?